assets/js/sections/contesting.js

```diff
@@ -208,20 +214,35 @@ $('#start_date').change(function () {
 
 // On Key up check and suggest callsigns
 $("#callsign").keyup(function () {
-	var call = $(this).val();
+	var call = $(this).val().toUpperCase();
 	if (call.length >= 3) {
 
-		$.ajax({
-			url: 'lookup/scp',
-			method: 'POST',
-			data: {
-				callsign: $(this).val().toUpperCase()
-			},
-			success: function (result) {
-				$('.callsign-suggestions').text(result);
-				highlight(call.toUpperCase());
-			}
-		});
+        if (scp_data.status < 0 || scp_data.request != call.substr(0, scp_data.request.length)) {
+          scp_data.status = 0;
+          scp_data.request = call;
+          scp_data.data = [];
+          $.ajax({
+              url: 'lookup/scp',
+              method: 'POST',
+              data: {
+                  callsign: call
+              },
+              success: function (result) {
+                  scp_data.status = 1;
+                  scp_data.data = result.split(" ");
+
+                  var call = $("#callsign").val().toUpperCase(); // Might have changed while running the query...
+                  $('.callsign-suggestions').text(scp_data.data.filter((el) => el.startsWith(call)).join(' '));
+                  highlight(call);
+              }
+          });
+
+        } else {
+          // Filter the already obtained list:
+          $('.callsign-suggestions').text(scp_data.data.filter((el) => el.startsWith(call)).join(' '));
```

Comment:
same thing here, we should add an existential and a default `[]`

Reply:
Actually I'm not sure anymore if this `startsWith` is correct: We may have prefixed callsigns in the result (type 'MM0' and have sth. like 'DL/MM0ASDF' in the results). I'll think about a regex.

Reply:
good point, maybe use `.includes`

```javascript
const string = "foo";
const substring = "oo";
console.log(string.includes(substring)); // true
```